pandas/tests/frame/test_quantile.py

```diff
@@ -467,3 +467,14 @@ def test_quantile_empty(self):
 
         # FIXME (gives NaNs instead of NaT in 0.18.1 or 0.19.0)
         # res = df.quantile(0.5, numeric_only=False)
+
+    def test_quantile_empty_no_columns(self):
+        # GH#23925 _get_numeric_data may drop all columns
+        df = pd.DataFrame(pd.date_range("1/1/18", periods=5))
+        result = df.quantile(0.5)
+        expected = pd.Series([], index=[], name=0.5)
```

Comment:
can you add a name to the index of the frame which I think should propagate here?

Reply:
good point.  i think it should be df.columns.name that propagates.

It looks like DataFrame_get_numeric_data isn't retaining the columns names, will open a separate issue

Reply:
also I pushed the 1.0 whatsnew if you rebase